econml/cate_estimator.py

```diff
@@ -456,6 +460,93 @@ def const_marginal_effect_inference(self, X=None):
         """
         pass
 
+    def shap_values(self, X, *, feature_names=None, treatment_names=None, output_names=None):
+        """ Shap value for the final stage models (const_marginal_effect)
+
+        Parameters
+        ----------
+        X: (m, d_x) matrix
+            Features for each sample. Should be in the same shape of fitted X in final stage.
+        feature_names: optional None or list of strings of length X.shape[1] (Default=None)
+            The names of input features.
+        treatment_names: optional None or list (Default=None)
+            The name of treatment. In discrete treatment scenario, the name should not include control name.
+        output_names:  optional None or list (Default=None)
+            The name of the outcome.
+
+        Returns
+        -------
+        shap_outs: nested dictionary of Explanation object
+            A nested dictionary by using each Y and each T as key and the shap_values explanation object as value.
+
+
+        """
+        d_t = self._d_t[0] if self._d_t else 1
```

Comment:
This naming of the dict seems to appear in multiple places. It should be written only once somewhere. Maybe in the shap.py file as a utility function.